arlas-core/src/main/java/io/arlas/server/utils/RequestColumnFilter.java

```diff
@@ -0,0 +1,153 @@
+            //no filter - always allowed
+            return true;
+        }
+
+        if (columnFilter.contains(field)) {
+            //field itself is allowed
+            return true;
+        }
+
+        String[] splitted = field.split("\\.");
+        if (splitted.length == 1) {
+            //field itself not allowed, and no parent field
+            return false;
+        }
+
+        //check if parent field is allowed
+        String withoutLastPart = String.join(".", Arrays.copyOfRange(splitted, 0, splitted.length - 1));
+        return isAllowed(columnFilter, withoutLastPart);
+    }
+
+    public static boolean isForbidden(Set<String> columnFilter, String field) {
+        return !RequestColumnFilter.isAllowed(columnFilter, field);
+    }
+
+    public static boolean isFilterDefined(Set<String> columnFilter) {
+        return !isNoFilterDefined(columnFilter);
+    }
+
+    public static boolean isNoFilterDefined(Set<String> columnFilter) {
+        return columnFilter.isEmpty();
+    }
+
+    /**
+     * Append the collection mandatory fields to the user' column filter
+     * @param columnFilter
+     * @param collectionReference
+     * @return
+     */
+    public static Set<String> getFullColumnFilter(Optional<String> columnFilter, CollectionReference collectionReference) {
+        Optional<Set<String>> columnFilterSet = columnFilter.filter(StringUtils::isNotBlank).map(cf -> Arrays.stream(cf.replaceAll("\\.\\*", "").replaceAll(" ", "").split(",")).collect(Collectors.toSet()));
+        //add collection mandatory fields, but only if a column filter is provided
+        columnFilterSet.map(cols -> cols.addAll(Arrays.asList(
+                collectionReference.params.idPath,
+                collectionReference.params.geometryPath,
+                collectionReference.params.centroidPath,
+                collectionReference.params.timestampPath)));
+
+        return columnFilterSet.orElse(new HashSet<>());
+    }
+
+    /**
+     * Check that there aren't forbidden fields into the requests
+     * @param userColumnFilter
+     * @param collectionReference
+     * @param basicRequest
+     * @param headerRequest
+     * @throws InternalServerErrorException if fields cannot be extracted from requests
+     * @throws ColumnUnavailableException if a field is forbidden
+     */
+    public static void checkRequest(Optional<String> userColumnFilter,
+                                    CollectionReference collectionReference,
+                                    Request basicRequest,
+                                    Request headerRequest)
+            throws InternalServerErrorException, ColumnUnavailableException {
+
+        Set<String> fullColumnFilter = getFullColumnFilter(userColumnFilter, collectionReference);
+        if (RequestColumnFilter.isNoFilterDefined(fullColumnFilter)) {
+            return;
+        }
+
+        checkQHasCol(basicRequest);
+        checkQHasCol(headerRequest);
+
```

Comment:
`collectionReference.collectionParams` has a `filter` attribute that should be checked it toot

Reply:
Can you cancel the comment if it is not relevant anymore?